source/tutorials/module-system/module-system.md

```diff
@@ -0,0 +1,1289 @@
+
+Now import this new `path.nix` module from your `marker.nix` module:
+
+```diff
+# marker.nix
+ in {
+
++  imports = [
++    ./path.nix
++  ];
++
+   options = {
+
+     users = lib.mkOption {
+```
+
+## The Arrival Marker
+
+Copy the `departure` option declaration to a new `arrival` option in `marker.nix`, to complete the initial path implementation:
+
+```diff
+# marker.nix
+         type = markerType;
+         default = {};
+       };
++
++      arrival = lib.mkOption {
++        type = markerType;
++        default = {};
++      };
+     };
+```
+
+Next, add an `arrival.style.label` attribute to the `config` block, mirroring the `departure.style.label`:
+
+```diff
+# marker.nix
+     config = {
+       departure.style.label = lib.mkDefault
+         (firstUpperAlnum name);
++      arrival.style.label = lib.mkDefault
++        (firstUpperAlnum name);
+     };
+   });
+```
+
+Finally, update the return list in the function passed to `concatMap` in `map.markers` to also include the `arrival` marker for each user:
+
+```diff
+# marker.nix
+     map.markers = lib.filter
+       (marker: marker.location != null)
+       (lib.concatMap (user: [
+-        user.departure
++        user.departure user.arrival
+       ]) (lib.attrValues config.users));
+
+     map.center = lib.mkIf
+```
+
+Now you have the basesis to define paths on the map, connecting pairs of departure and arrival points.
+
+## Connecting Markers by Paths
+
+In the path module, define a path connecting every user's departure and arrival locations:
+
+```diff
+# path.nix
+   config = {
++
++    map.paths = builtins.map (user: {
++      locations = [
++        user.departure.location
++        user.arrival.location
++      ];
++    }) (lib.filter (user:
++      user.departure.location != null
++      && user.arrival.location != null
++    ) (lib.attrValues config.users));
++
+     requestParams = let
+       attrForLocation = loc:
+         "$(geocode ${lib.escapeShellArg loc})";
+```
+
+The new `map.paths` attribute contains a list of all valid paths defined for all users.
+
+A path is valid only if the `departure` and `arrival` attributes are set for that user.
+
+## Path Styling: Weight
+
+Your users have spoken, and they demand the ability to customize the styles of their paths with a `weight` option.
+
+As before, you'll now declare a new submodule for the path style.
+
+While you could also directly declare the `style.weight` option, in this case you should use the submodule to be able reuse the path style type later.
+
+Add the `pathStyleType` submodule option to the `let` block in `path.nix`:
+```diff
+# path.nix
+ { lib, config, ... }:
+ let
++
++  pathStyleType = lib.types.submodule {
++    options = {
++      weight = lib.mkOption {
++        type = lib.types.ints.between 1 20;
++        default = 5;
++      };
++    };
++  };
++
+   pathType = lib.types.submodule {
+```
+
+:::{note}
+The `ints.between <lower> <upper>` type allows integers in the given (inclusive) range.
```

Comment:
This should be part of the normal flow of the text, similar to every other type we introduced.